lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -2355,23 +2355,21 @@ bool TypeChecker::typeCheckBinding(
   if (!initializer->getType())
     initializer->setType(ErrorType::get(Context));
 
-  // If the type of the pattern is inferred, assign error types to the pattern
-  // and its variables, to prevent it from being referenced by the constraint
-  // system.
+  // Assign error types to the pattern and its variables, to prevent it from
+  // being referenced by the constraint system.
   if (patternType->hasUnresolvedType() ||
-      patternType->hasUnboundGenericType()) {
+      patternType->hasUnboundGenericType())
     pattern->setType(ErrorType::get(Context));
```

Comment:
Could you brace this? It's quite confusing to read.

Reply:
Sure done.